### PR TITLE
[Fix-5781][UT] Fix test coverage in sonar

### PR DESCRIPTION
**Pull request by shink**
## Purpose of the pull request

This pull request makes `jacoco` running in `offline-instrumentation` to get coverage. This way classes get instrumented by jacoco before any runtime modification can take place.

For details, please see: #5781

## Brief change log

- Make jacoco running in offline-instrumentation

## Verify this pull request

If this pull request works, the coverage of tests that have already been covered will not be %0 in [SonarCloud](https://sonarcloud.io/code?id=apache-dolphinscheduler).

### CODE REVIEW

**Comment by CalvinKirs** [comment]
hi, Can I review this PR?

**Comment by shink** [comment]
@CalvinKirs Sure. But this [workflow](https://github.com/apache/dolphinscheduler/pull/5817/checks?check_run_id=3155366037) failed. I tested in another branch with the same changes and all workflows were successful. Please see: https://github.com/shink/dolphinscheduler/actions/runs/1064891274
I will check again. Please review when all the workflows are successful. Thank you very much.

File: dolphinscheduler-microbench/pom.xml [hunk 1]
```diff
@@ -57,6 +57,13 @@
             <scope>compile</scope>
         </dependency>
 
+        <dependency>
+            <groupId>org.jacoco</groupId>
+            <artifactId>org.jacoco.agent</artifactId>
+            <classifier>runtime</classifier>
+            <scope>test</scope>
+        </dependency>
+
```
Comment (CalvinKirs) [inline dolphinscheduler-microbench/pom.xml]: this model usually needn't sonar check

Comment (shink) [reply dolphinscheduler-microbench/pom.xml]: OK. Thank you. I will revise it later.

**Comment by Jave-Chen** [comment]
Good job

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**